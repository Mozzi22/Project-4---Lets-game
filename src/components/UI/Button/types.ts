import React from 'react';
import { colors } from 'src/components/UI/baseLayout';

export interface IBtn {
    margin?: string;
    padding?: string;
    width?: string;
    height?: string;
    cursor?: string;
    bgColor?: string;
    borderRadius?: string;
    border?: string;
    outline?: string;
    transition?: string;
    fontSize?: string;
    focusColor?: string;
    bgColorDisabled?: string;
    theme: 'dark' | 'light';
    colors: typeof colors;
}
export interface IButton{
    id: string;
    name?: string;
    onClick: (e : React.MouseEvent<HTMLElement, MouseEvent>) => void;
    title?: string;
    type?: 'button';
    value?: string;
    borderRadius?: string;
    isDisabled?: boolean;
    color?: string;
    fontSize?: string;
    fontWeight?: string;
    bgColorDisabled?: string;
    bgColor?: string;
    bgColorHover?: string;
    height?: string;
    width?: string;
    padding?: string;
    margin?: string;
    content: string;
    transition?: string;
    focusColor?: string;
}
