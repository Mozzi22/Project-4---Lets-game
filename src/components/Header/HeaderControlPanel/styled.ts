import styled from 'styled-components';

export const StControl = styled.div`
        display: flex;
        justify-content: flex-end;
        align-items: center;
        height: 100%;
        width: 80%;
        margin-right: 20px;
        p { color: #fff }
`;
