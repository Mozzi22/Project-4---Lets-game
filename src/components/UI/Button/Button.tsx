import React from 'react';
import { useTranslation } from 'react-i18next';
import { useTheme } from '../../Hooks/useTheme';
import { StButton } from './styled';
import { IButton } from './types';

const Button = ({
    id,
    name,
    onClick,
    title,
    type = 'button',
    value,
    borderRadius,
    isDisabled,
    color,
    fontSize,
    bgColorDisabled,
    bgColor,
    backgroundImage,
    backgroundSize,
    height,
    width,
    padding,
    margin,
    content,
    transition,
}: IButton) => {
    const { t } = useTranslation();
    const { colors, theme } = useTheme();
    return (
        <StButton
            id={id}
            name={name}
            onClick={onClick}
            type={type}
            value={value}
            title={title}
            bgColorDisabled={bgColorDisabled}
            backgroundImage={backgroundImage}
            backgroundSize={backgroundSize}
            borderRadius={borderRadius}
            disabled={isDisabled}
            color={color}
            fontSize={fontSize}
            bgColor={bgColor}
            height={height}
            width={width}
            padding={padding}
            margin={margin}
            transition={transition}
            theme={theme}
            colors={colors}
        >
            {t(content)}
        </StButton>
    );
};

export default Button;
