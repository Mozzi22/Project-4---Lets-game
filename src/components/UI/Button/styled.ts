import styled from 'styled-components';
import {
  colorDefault,
  focusColorDefault,
  disabledColor,
  bgColorDisabledDefault,
  disabledBorder,
  hoverBgDefault,
} from '../baseLayout';
import { IBtn } from './types';

export const StButton = styled.button < IBtn > `
  ${({ margin = '0 0 3px 0' }) => margin && `margin: ${margin}`};
  font-family: 'Oxanium', cursive;
  padding: ${({ padding = '5px' }) => padding};
  width: ${({ width = '170px' }) => width};
  height: ${({ height = '35px' }) => height};
  ${({ cursor = 'pointer' }) => cursor && `cursor: ${cursor}`};
  color: ${({ color = colorDefault }) => color};
  border-radius: ${({ borderRadius = '7px' }) => borderRadius};
  border: ${({ border = 'none' }) => border};
  outline: ${({ outline = 'none' }) => outline};
  ${({ transition = 'all 300ms ease-in-out' }) => transition && `transition: ${transition}`};
  font-size: ${({ fontSize = '16px' }) => fontSize};
  &:hover {
    border: ${({ border = 'none' }) => border};
    }
    background-color: ${({ bgColor = hoverBgDefault }) => bgColor};
  &:focus {
    color: ${({ focusColor = focusColorDefault }) => focusColor};
    background-color: ${({ bgColor = hoverBgDefault }) => bgColor};
    }
  &:active{
    border: ${({ border = disabledBorder }) => border};
    background-color: ${({ bgColorDisabled = '#00a6ff' }) => bgColorDisabled};
    }
  &:disabled{
    border: ${({ border = disabledBorder }) => border};
    background-color: ${({ bgColorDisabled = bgColorDisabledDefault }) => bgColorDisabled};
    color: ${({ focusColor = disabledColor }) => focusColor};
    }
`;
