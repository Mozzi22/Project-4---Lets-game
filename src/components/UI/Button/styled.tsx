import styled from 'styled-components';
import { IBtn } from './types';
import {
  bgColorDefault,
  colorDefault,
  focusColorDefault,
  disabledColor,
  bgColorDisabledDefault,
  disabledBorder,
  hoverBgDefault,
} from '../baseLayout';

export const StButton = styled.button < IBtn > `
  ${({ margin = '0 0 3px 0' }) => margin && `margin: ${margin}`};
  font-family: 'Play', sans-serif;
  padding: ${({ padding = '5px' }) => padding};
  width: ${({ width = '170px' }) => width};
  height: ${({ height = '35px' }) => height};
  ${({ cursor = 'pointer' }) => cursor && `cursor: ${cursor}`};
  color: ${({ color = colorDefault }) => color};
  background-color: ${({ bgColor = bgColorDefault }) => bgColor};
  border-radius: ${({ borderRadius = '7px' }) => borderRadius};
  border: ${({ border = 'none' }) => border};
  outline: ${({ outline = 'none' }) => outline};
  ${({ transition = 'all 300ms ease-in-out' }) => transition && `transition: ${transition}`};
  font-size: ${({ fontSize = '16px' }) => fontSize};
  &:hover {
    border: ${({ border = 'none' }) => border};
    }
    background-color: ${({ bgColor = hoverBgDefault }) => bgColor};
  &:focus {
    color: ${({ focusColor = focusColorDefault }) => focusColor};
    background-color: ${({ bgColor = hoverBgDefault }) => bgColor};
    }
  &:active{
    border: ${({ border = disabledBorder }) => border};
    background-color: ${({ bgColorDisabled = '#00a6ff' }) => bgColorDisabled};
    }
  &:disabled{
    border: ${({ border = disabledBorder }) => border};
    background-color: ${({ bgColorDisabled = bgColorDisabledDefault }) => bgColorDisabled};
    color: ${({ focusColor = disabledColor }) => focusColor};
    }
`;
