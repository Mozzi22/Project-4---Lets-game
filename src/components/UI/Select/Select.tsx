import React from 'react';
import { useTranslation } from 'react-i18next';
import { useTheme } from 'src/components/Hooks/useTheme';
import { StSelect, StOption } from './styled';
import { ISelect } from './types';

const Select: React.FC<ISelect> = ({
    id,
    onClick,
    onChange,
    content,
    name,
    value,
    margin,
    padding,
    width,
    height,
    cursor,
    bgColor,
    borderRadius,
    border,
    outline,
    transition,
    fontSize,
    color,
    focusColor,
    bgColorDisabled,
    options,
    defaultValue,
    selected,
    }) => {
    const { t } = useTranslation();
    const { colors, theme } = useTheme();
    return (
        <StSelect
            id={id}
            onClick={onClick}
            onChange={onChange}
            content={content}
            name={name}
            value={value}
            borderRadius={borderRadius}
            color={color}
            border={border}
            outline={outline}
            bgColorDisabled={bgColorDisabled}
            focusColor={focusColor}
            fontSize={fontSize}
            bgColor={bgColor}
            cursor={cursor}
            height={height}
            width={width}
            padding={padding}
            margin={margin}
            transition={transition}
            options={options}
            defaultValue={defaultValue}
            theme={theme}
            colors={colors}
        >
            {options.map(el => (
                <StOption
                    id={id}
                    value={el.value}
                    key={el.id}
                    onClick={onClick}
                    selected={selected}
                    bgColor={bgColor}
                >
                    {el.label}
                        {t(content)}
                </StOption>
                ))
            }
        </StSelect>
    );
};

export default Select;
