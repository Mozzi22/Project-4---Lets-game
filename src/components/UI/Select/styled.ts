import styled from 'styled-components';

import {
  disabledColor,
  bgColorDisabledDefault,
  disabledBorder,
  activeBorder,
} from '../baseLayout';
import { ISelect, IOption } from './types';

export const StSelect = styled.select < ISelect > `
  ${({ margin = '0 0 3px 0' }) => margin && `margin: ${margin}`};
  font-family: 'Poppins', sans-serif;
  padding: ${({ padding = '0 0 0 5px' }) => padding};
  width: ${({ width = '170px' }) => width};
  height: ${({ height = '30px' }) => height};
  ${({ cursor = 'pointer' }) => cursor && `cursor: ${cursor}`};
  color: ${({ colors, theme }) => colors[theme].colorTextForm};
  background-color: ${({ colors, theme }) => colors[theme].backgroundSelect};
  border-radius: ${({ borderRadius = '7px' }) => borderRadius};
  border: ${({ border = activeBorder }) => border};
  outline: ${({ outline = 'none' }) => outline};
  ${({ transition = 'all 300ms ease-in-out' }) => transition && `transition: ${transition}`};
  font-size: ${({ fontSize = '16px' }) => fontSize};
  option {
      background-color: ${({ colors, theme }) => colors[theme].backgroundSelect};
  }
  &:disabled{
    border: ${({ border = disabledBorder }) => border};
    background-color: ${({ bgColorDisabled = bgColorDisabledDefault }) => bgColorDisabled};
    color: ${({ focusColor = disabledColor }) => focusColor};
    }
`;

export const StOption = styled.option < IOption > `
  font-family: 'Poppins', sans-serif;
  font-size: 16px;
`;
