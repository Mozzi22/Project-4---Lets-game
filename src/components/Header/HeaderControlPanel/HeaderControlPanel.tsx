import React, { useContext } from 'react';
import { useTranslation } from 'react-i18next';
import Button from 'src/components/UI/Button';
import Select from 'src/components/UI/Select';
import { Theme } from 'src/components/Hocs/withTheme';
import { options } from 'src/constants/ui';
import { StControl } from './styled';

const HeaderControlPanel = () => {
    const { i18n } = useTranslation();
    const { theme, changeTheme } = useContext(Theme);
    const [selected, setSelected] = React.useState('');

    React.useEffect(() => {
        const lastSelected = localStorage.getItem('lang');
        setSelected(lastSelected);
    }, [localStorage.getItem('lang')]);

    const onChangeLanguage = (lng, i18n) => {
        i18n.changeLanguage(lng);
        localStorage.setItem('lang', lng);
    };

    const handleChangeLanguage = (e) => {
        onChangeLanguage(e.target.value, i18n);
    };

    const toggleThemeMode = () => changeTheme(
        theme === 'dark' ? 'light' : 'dark',
    );

    return (
        <StControl>
            <Button
                id='theme-btn'
                fontSize='26px'
                width='40px'
                height='40px'
                margin='0 15px 0'
                borderRadius='0px'
                onClick={toggleThemeMode}
                bgColor='transparent'
                backgroundSize='cover'
                hoverColor='transparent'
                bgColorDisabled='transparent'
            />
            <Select
                id={options.id}
                options={options}
                onChange={handleChangeLanguage}
                value={selected || ''}
                width='70px'
                bgColor='#b16fdd'
            />
        </StControl>
    );
};

export default HeaderControlPanel;
