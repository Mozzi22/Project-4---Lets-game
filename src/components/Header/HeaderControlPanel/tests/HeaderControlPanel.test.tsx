import React from 'react';
import { shallow, mount } from 'enzyme';
import Button from 'src/components/UI/Button';
import Select from 'src/components/UI/Select';
import HeaderControlPanel from '../HeaderControlPanel';

jest.mock('react-i18next', () => ({
  useTranslation: jest
    .fn()
    .mockReturnValue({ i18n: { changeLanguage: jest.fn() }, t: jest.fn() }),
}));

describe('HeaderControlPanel', () => {
  it('Should match snapshot', () => {
    const component = shallow(<HeaderControlPanel/>);
    expect(component.html()).toMatchSnapshot();
  });
  it('Should have Button', () => {
    const component = mount(<Button />);
    expect(component.find('Button')).toHaveLength(1);
    });
  it('Should have Select', () => {
    const component = mount(<Select />);
    expect(component.find('Select')).toHaveLength(1);
  });
});
