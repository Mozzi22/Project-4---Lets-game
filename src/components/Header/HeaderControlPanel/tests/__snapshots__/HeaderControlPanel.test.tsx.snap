// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderControlPanel Should match snapshot 1`] = `"<div class=\\"styled__StControl-sc-1lhkoni-0 hdTeNn\\"><button id=\\"theme-btn\\" type=\\"button\\" font-size=\\"26px\\" height=\\"40px\\" width=\\"40px\\" class=\\"styled__StButton-sc-15atqtm-0 fbkdJe\\"></button><select width=\\"70px\\" class=\\"styled__StSelect-sc-1hj7ohk-0 gkgfCL\\"><option id=\\"1\\" value=\\"en\\" class=\\"styled__StOption-sc-1hj7ohk-1 iWufHR\\">en</option><option id=\\"2\\" value=\\"ru\\" class=\\"styled__StOption-sc-1hj7ohk-1 iWufHR\\">ru</option></select></div>"`;
